src/NHibernate/Util/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Reflection;
+using NHibernate.Bytecode;
+
+namespace NHibernate.Util
+{
+	internal static class ServiceProviderExtensions
+	{
+		/// <summary>
+		/// Get a service, throwing if it cannot be resolved.
+		/// </summary>
+		/// <param name="serviceProvider">The service provider.</param>
+		/// <param name="serviceType">The service interface, base class or concrete implementation class.</param>
+		/// <returns>The service instance.</returns>
+		/// <exception cref="ArgumentNullException">thrown if <paramref name="serviceType"/> is <see langword="null" />.</exception>
+		/// <exception cref="HibernateServiceProviderException">thrown if the service cannot be resolved.</exception>
+		public static object GetInstance(this IServiceProvider serviceProvider, System.Type serviceType)
```

Comment:
Maybe this should be renamed `GetMandatoryService`. Just from their current name, `GetInstance` vs `GetService` semantic is not obvious.

Reply:
Renamed.